Modix.Services/Notes/NoteCreatorService.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Modix.Services.Notes
+{
+    using System;
+    using System.Threading.Tasks;
+    using Data;
+    using Data.Models.Moderation;
+    using Discord;
+    using Serilog;
+
+    public class NoteCreatorService : INoteCreatorService
+    {
+        private readonly ModixContext _modixContext;
+
+        public NoteCreatorService(ModixContext modixContext)
+        {
+            _modixContext = modixContext;
+        }
+
+        public async Task<bool> Create(IGuildUser user, string message, string recordedBy)
```

Comment:
I'm not a huge fan of swallowing errors and returning a `bool` to indicate success or failure. Any reason to not just let the exception bubble up?

Reply:
We should check to make sure that these aren't `null` or empty values. If `recordedBy` or `user` aren't provided it should be an error. If `message` is empty we should probably just make it a no-op and immediately return `true`.

Reply:
It's just a personal preference thing for me. You either successfully create a Note or you don't. The error is caught in the service and logged so I just thought in this case that would be enough.

If the service returns false in any way we replay a message to the user that there has been an error.

Reply:
Long term, what I want to add is an `IServiceAction` system, basically an abstraction of `IHttpAction`, so error handling can be both granular, and not exception-based. For now, I've just been doing exception-based errors, except when the next level up might legitimately expect the error and change its behavior, accordingly.